[Bugfix] Bill payment exchange rate (#155)
fivetran/dbt_quickbooks: CHANGELOG.md

# dbt_quickbooks v0.17.2
[PR #155](https://github.com/fivetran/dbt_quickbooks/pull/155) introduces the following updates:

## Bug Fixes

> - Modified `int_quickbooks__bill_payment_double_entry model` to ensuree bill payment accuracy by updating the `converted_amount` logic:

Reviewer: ```suggestion
- Modified `int_quickbooks__bill_payment_double_entry model` to ensure bill payment accuracy by updating the `converted_amount` logic. These changes improve the integrity and granularity of debit/credit entries for bill payments:
```

Author: Updated

>   - Utilized the line-level `amount` from bill payment lines instead of `total_amount` from bill payments.

Reviewer: Can you add _why_ you applied this change.

Author: Added.

  - Applied `exchange_rate` from the associated bill where available, falling back to the bill payment exchange rate otherwise.
  - Inverted the sign of the `amount` when associated with a journal entry (to correctly reflect the accounting direction).
  - Updated joins to link bill payments with related bills and bill payment lines via the `bill_linked_txn` table to support more accurate allocation of payment amounts.

> - These changes improve the integrity and granularity of debit/credit entries for bill payments.

Reviewer: ```suggestion
```
It reads strange that this is it's own bullet. Let's instead add this to the first bullet and let all the sub-bullets specify what the integrity changes are.

Author: Moved around.

> # dbt_quickbooks v0.17.2

Author: Sounds good, I've changed this.

# dbt_quickbooks v0.18.0
This release introduces the following updates.

> ## Bug Fixes

Reviewer 2: I would mark this as breaking changes since it's called out at the end of the section.

Author: Ah good catch!

- Modified `int_quickbooks__bill_payment_double_entry model` to ensure bill payment accuracy by updating the `converted_amount` logic. These changes improve the integrity and granularity of debit/credit entries for bill payments. ([PR #155](https://github.com/fivetran/dbt_quickbooks/pull/155))

>   - Utilized the line-level `amount` from bill payment lines instead of `total_amount` from bill payments, then aggregated that upward. Bill payment lines in a bill payment can have different exchange rates applied depending on which bill it's associated with, so the bill exchange rate can come into play.

Reviewer 2: > then aggregated that upward

Could you clarify what this means? Does that mean something like summing the lines to calculate the total or something else?

Author: Yep essentially! Updated.

  - Applied `exchange_rate` from the associated bill where available, falling back to the bill payment exchange rate otherwise.
  - Inverted the sign of the `amount` when associated with a journal entry (to correctly reflect the accounting direction).
  - Updated joins to link bill payments with related bills and bill payment lines via the `bill_linked_txn` table to support more accurate allocation of payment amounts.

> - We've made this a **breaking change** as this update could potentially change (but fix) the amount outputs of the end models.

Reviewer 2: I would move this to the top.

Author: Moved.

## Under the Hood

> - Added general ledger consistency test to ensure matching records between production and development models. ([#155](https://github.com/fivetran/dbt_quickbooks/pull/155))

Reviewer 2: ```suggestion
- Added a general ledger consistency test to ensure matching records between production and development models. ([#155](https://github.com/fivetran/dbt_quickbooks/pull/155))
```

Author: Committed.
